pkg/virt-api/rest/subresource.go

```diff
@@ -677,6 +677,32 @@ func (app *SubresourceAPIApp) UnfreezeVMIRequestHandler(request *restful.Request
 
 }
 
+func (app *SubresourceAPIApp) SoftRebootVMIRequestHandler(request *restful.Request, response *restful.Response) {
+
+	log.Log.Info("SoftRebootVMIRequestHandler")
+	validate := func(vmi *v1.VirtualMachineInstance) *errors.StatusError {
+		if vmi.Status.Phase != v1.Running {
+			return errors.NewConflict(v1.Resource("virtualmachineinstance"), vmi.Name, fmt.Errorf("VM is not running"))
+		}
+		condManager := controller.NewVirtualMachineInstanceConditionManager()
+		if condManager.HasConditionWithStatus(vmi, v1.VirtualMachineInstancePaused, v12.ConditionTrue) {
+			return errors.NewConflict(v1.Resource("virtualmachineinstance"), vmi.Name, fmt.Errorf("VMI is paused"))
+		}
+		if !condManager.HasCondition(vmi, v1.VirtualMachineInstanceAgentConnected) {
+			if vmi.Spec.Domain.Features != nil && !(*vmi.Spec.Domain.Features.ACPI.Enabled) {
```

Comment:
I think the conditions when `vmi.Spec.Domain.Features == nil` and `Spec.Domain.Features.ACPI.Enabled == nil`  should also be checked Maybe smth like below:
```go
if features := vmi.Spec.Domain.Features; features == nil || features.ACPI.Enabled == nil || !(*features.ACPI.Enabled) {
    return errors.NewConflict(v1.Resource("virtualmachineinstance"), vmi.Name, fmt.Errorf("VMI neither have the agent connected nor the ACPI feature enabled"))
}
```

Reply:
@vasiliy-ul re: the above code example, note that ACPI [default to enabled](https://github.com/kubevirt/kubevirt/blob/e622b99f6cfe4cbd659ff408dffb411effa2664c/staging/src/kubevirt.io/client-go/apis/core/v1/schema.go#L1045), so `features == nil || features.ACPI.Enabled == nil` implies ACPI enabled ==> no error.

Reply:
ah... seems that I missed the 'default to enabled' part. so I did mess it up eventually :) sorry, my bad. thanks for noting it.

Reply:
updated